### ISSUE TITLE
bug with lines for large values on x axis when using xlims!

### ISSUE THREAD

**Issue by NilsNiggemann** [MakieOrg/Makie.jl#3133]
When a line is plotted with relatively large values on the x axis it sometimes displays an erronous line that seems to start at x=0.
Here is a MWE:

```
x = [100000.,1000000.] # does not contain 0.
y = [0.,0.]

lines(x,y,color =  :red) # this line shouldnt be visible!
xlims!(0,1)
current_figure()
```
- Scattering the same data points produces the expected behavior (no data visible in plot)
- The bug depends on the values of x chosen, for example multiplying x by 2 gives correct results (for some reason unclear to me). 

____________________________________________________________________
- [x] are you running newest version (version from docs) ?
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- - I am on the current version of CairoMakie v0.10.7
- What platform + GPU are you on?
```
  Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 4 × Intel(R) Core(TM) i5-4690 CPU @ 3.50GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, haswell)
  Threads: 4 on 4 virtual cores
```
-GPU  
```
       *-display                 
       description: VGA compatible controller
       product: Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics Controller
       vendor: Intel Corporation
       physical id: 2
       bus info: pci@0000:00:02.0
       version: 06
       width: 64 bits
       clock: 33MHz
       capabilities: vga_controller bus_master cap_list rom
       configuration: driver=i915 latency=0
       resources: irq:32 memory:f7800000-f7bfffff memory:e0000000-efffffff ioport:f000(size=64) memory:c0000-dffff
WARNING: output may be incomplete or inaccurate, you should run this program as super-user.
```
(I don't have root permissions here, but I don't expect the GPU to be relevant anyway)

**Comment by jkrumbiegel** [comment]
Interesting, this seems to be a bug in Cairo. Because I checked what positions get computed for the line vertices in CairoMakie and that's `projected_positions = Vec{2, Float32}[[7.36f7, 284.0], [7.36f8, 284.0]]`. But in the svg that gets rendered, I get `d="M 6491184 300 L -2197456 300 "` as a path, and the `L -2197456` suggests that some integer wraparound has happened. I'm not sure which part of Cairo it is that cannot deal with very large points far outside the canvas boundaries.

**Comment by ffreyer** [comment]
This specific case is fine for me on breaking-0.21, but it might have just moved to different values

**Comment by ffreyer** [comment]
I assume this got fixed by either Float64 changes or #3958 updating the lines clipping code in CairoMakie
